src/main/java/io/papermc/lib/features/bedspawnlocation/BedSpawnLocationPaper.java

```diff
@@ -0,0 +1,24 @@
+package io.papermc.lib.features.bedspawnlocation;
+
+import io.papermc.lib.PaperLib;
+import org.bukkit.Location;
+import org.bukkit.entity.Player;
+
+import java.util.concurrent.CompletableFuture;
+
+public class BedSpawnLocationPaper implements BedSpawnLocation {
+
+    @Override
+    public CompletableFuture<Location> getBedSpawnLocationAsync(Player player, boolean isUrgent) {
+        Location bedLocation = player.getPotentialBedLocation();
+        if (bedLocation == null) {
+            return CompletableFuture.completedFuture(null);
+        }
+        CompletableFuture<Location> future = new CompletableFuture<>();
+        PaperLib.getChunkAtAsync(bedLocation.getWorld(), bedLocation.getBlockX(), bedLocation.getBlockZ(), false, isUrgent).thenAccept(chunk -> future.complete(player.getBedSpawnLocation())).exceptionally(ex -> {
```

Comment:
you missed >> 4 on X/Z here.

also don't do the manual future

just use 
.thenCompose((chunk) -> player.getBedSpawnLocation());